File: 7KYU/Scaling Squared Strings/solution.clj
(ns scalesqstrings.core)
(require '[clojure.string :as str])

(defn scale [s h v]
  (if (= "" s)
    ""
    (->> (str/split-lines s)
         (map #(str/split % #""))
         (map (fn [x] (mapcat (fn [y] (repeat h y)) x)))
         (mapcat #(repeat v %))
         (map (partial apply str))
         (str/join "\n"))))

File: 7KYU/Scaling Squared Strings/solution.go
package kata

import "strings"

func Scale(s string, k, v int) string {

	if s == "" {
		return ""
	}

	lines := strings.Split(s, "\n")
	res := []string{}

	for _, l := range lines {
		s := ""
		for _, c := range l {
			s += strings.Repeat(string(c), k)
		}
		for i := 0; i < v; i++ {
			res = append(res, s)
		}
	}

	return strings.Join(res, "\n")
}
